subscription/src/main/java/org/killbill/billing/subscription/api/svcs/DefaultSubscriptionInternalApi.java

```diff
@@ -275,6 +277,99 @@ public SubscriptionBase apply(final DefaultSubscriptionBase input) {
         }
     }
 
+    @Override
+    public List<SubscriptionBase> createBaseSubscriptionsWithAddOns(final UUID accountId, final Iterable<BaseEntitlementWithAddOnsSpecifier> baseEntitlementWithAddOnsSpecifier, final InternalCallContext context) throws SubscriptionBaseApiException {
+        try {
+            final Catalog catalog = catalogService.getFullCatalog(true, true, context);
+            final CallContext callContext = internalCallContextFactory.createCallContext(context);
+
+            final List<SubscriptionAndAddOnsSpecifier> subscriptionAndAddOns = new ArrayList<SubscriptionAndAddOnsSpecifier>();
+            for (BaseEntitlementWithAddOnsSpecifier entitlementWithAddOnsSpecifier : baseEntitlementWithAddOnsSpecifier) {
+                final List<SubscriptionSpecifier> subscriptions = new ArrayList<SubscriptionSpecifier>();
+                final DateTime now = clock.getUTCNow();
+                final DateTime effectiveDate = (entitlementWithAddOnsSpecifier.getEntitlementEffectiveDate() != null) ?
+                                               DefaultClock.truncateMs(entitlementWithAddOnsSpecifier.getEntitlementEffectiveDate().toDateTimeAtCurrentTime()) : now;
+
+                final SubscriptionBaseBundle bundle = createBundleForAccount(accountId, entitlementWithAddOnsSpecifier.getExternalKey(), context);
+                final List<SubscriptionBase> subscriptionWithAddOns = createBaseSubscriptionWithAddOns(bundle.getId(), entitlementWithAddOnsSpecifier.getEntitlementSpecifier(),
+                                                                                                       entitlementWithAddOnsSpecifier.getEntitlementEffectiveDate().toDateTimeAtCurrentTime(),
+                                                                                                       entitlementWithAddOnsSpecifier.isMigrated(), context);
+                boolean first = true;
+                for (EntitlementSpecifier entitlement : entitlementWithAddOnsSpecifier.getEntitlementSpecifier()) {
```

Comment:
Do you need these checks below? Aren't they done by `createBaseSubscriptionWithAddOns` above? (I'm just trying to find a way to minimize duplicated code)